datafusion/core/src/sql/planner.rs

```diff
@@ -443,16 +463,21 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
         &self,
         t: TableWithJoins,
         ctes: &mut HashMap<String, LogicalPlan>,
+        outer_schema: Option<&DFSchema>,
```

Comment:
Given that `outer` means something already in terms of joins, I wonder if a name like `outer_query_schema` might potentially cause less confusion

Reply:
That makes sense. I renamed this.